src/engraving/rendering/dev/paint.cpp

```diff
@@ -122,6 +122,10 @@ void Paint::paintScore(Painter* painter, Score* score, const IScoreRenderer::Pai
                 painter->translate(-pageRect.topLeft());
             }
 
+            if (drawRect.bottom() > pageRect.bottom()) {
+                painter->setClipRect(pageRect);
```

Comment:
Makes sense to treat this as a special case, as there's really the only one way elements can end up being drawn "off" the page, and in that case there's no point trying to maintain the QtQuick dirty rectangle optimisations (which are what control the existing clipping rect).